### DISCUSSION TITLE
Is there a way to replace the DLL while Revit is open during the build process?

### DISCUSSION THREAD

**Discussion by cbptamtom** [Nice3point/RevitTemplates#49] [Q&A] [answered]
Thank you for an amazing open-source project. It helps in configuring code in a professional manner and makes it easier to extend the code. However, I have a problem when running/debugging as opening and closing Revit takes quite a lot of time during the coding process. Could you please help me with one or a few approaches to work more quickly and efficiently without having to open/close Revit continuously when running/debugging? When I first started with a standalone DLL to access the Revit API, I used to run the DLL through the Add-in Manager. But when configuring with Host/DI for ExternalApplication, I can’t use that approach because the Host is executing in Application.cs, inheriting ExternalApplication.

**Comment by Nice3point** [comment, marked as answer]
Many thanks) Using Host/DI the only option is to use Hot Reload or restart Revit. Hot Reload covers most cases. Starting with .Net 8 (Revit 2025) you can create a container in AssemblyLoadContext and configure its unloading. Another option is to set a dynamic name when building so Revit sees it as a new plugin

**Reply by cbptamtom** [reply]
Many thanks Roman for responding to my answer. Is there a pathway to start learning these things, such as where to begin to understand them, a course, or a sample project? Please guide me, as I am not a programmer by background; I am a civil engineer. I am also searching and learning about Hot Reload as you referred. I also tried to set a Dynamo naming but was not successful.

**Reply by cbptamtom** [reply]
![image](https://github.com/Nice3point/RevitTemplates/assets/133786016/6bd1b6b3-4333-4cbb-9f6f-07ddc3832a44)
Yeah, I just tried the Hot Reload feature of Rider and it's working well as I expected. I really appreciate it. Hopefully, you will introduce more methods in the future.
